modules/lib/src/main/resources/assets/js/app/ui2/list/CheckableListItemWithStatus.tsx

```diff
@@ -7,13 +8,16 @@ import {ContentIcon} from './ContentIcon';
 
 type Props = {
     content: ContentSummaryAndCompareStatus;
+    contentViewer: ContentSummaryViewer;
 } & Pick<SelectableListItemProps, 'className' | 'readOnly' | 'onCheckedChange' | 'checked'>
 
-const CheckableListItemWithStatusComponent = ({content, ...props}: Props): JSX.Element => {
+const CheckableListItemWithStatusComponent = ({content, contentViewer, ...props}: Props): JSX.Element => {
 
     const label = content.getPath().toString();
+    const summary = content.getContentSummary();
+    const imageUrl = summary ? contentViewer.resolveIconUrl(summary) : null;
     const status = CompareStatusFormatter.formatStatusText(content.getCompareStatus());
-    const icon = <ContentIcon contentType={content.getType()}/>;
+    const icon = <ContentIcon contentType={content.getType()} imageUrl={imageUrl}/>;
```

Comment:
Looking at this, I think it makes more sense to handle the image inside ContentIcon, since it may be used in multiple places. We don’t want to duplicate the URL-resolving logic.